### PR TITLE
Make it easier to add support for new template types

### PR DESCRIPTION
**Pull request by thekindofme**
This work will allow us to easily add support for new template types, such as YML (as requested in https://github.com/envato/stack_master/issues/81)

### CODE REVIEW

File: lib/stack_master/template_compiler.rb [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-        template_body = File.read(template_file_path)
-        if template_body.size > MAX_TEMPLATE_SIZE
-          # Parse the json and rewrite compressed
-          JSON.dump(JSON.parse(template_body))
-        else
-          template_body
-        end
-      end
+    def self.file_ext(template_file_path)
+      File.extname(template_file_path).gsub('.', '').to_sym
```
Comment (stevehodgkiss) [inline lib/stack_master/template_compiler.rb]: What about a file name with multiple extensions? Would it be useful to support that? `my_template.erb.json` for example.

**Comment by stevehodgkiss** [comment]
Much better :+1:

File: lib/stack_master/template_compiler.rb (hunk 1 above)
Comment (thekindofme) [inline lib/stack_master/template_compiler.rb]: We could add support for that if we need it, in the future? (YAGNI) or we could do it now. What do you think?

Comment (stevehodgkiss) [inline lib/stack_master/template_compiler.rb]: Let's do it later if we need it.

**Pull request merged**